### ISSUE TITLE
RuntimeError: d.is_cuda() INTERNAL ASSERT FAILED

### ISSUE THREAD

**Issue by wenboweiwb** [facebookresearch/detectron2#5331]
Every time I train Mask2Former with Detectron2, I encounter the same error. The specific error is as follows:

Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: d.is_cuda() INTERNAL ASSERT FAILED at "/opt/conda/conda-bld/pytorch_1678402421473/work/c10/cuda/impl/CUDAGuardImpl.h":31, please report a bug to PyTorch.

This error does not appear at the beginning but usually occurs after running for a long time, such as 5 hours. This makes debugging very difficult. I would be very grateful if someone could help me look into this issue.

**Comment by dunyang97** [comment]
do you have some solution about it?  I also face the totally some error, and i test several different version of cuda, torch, nvidia-driver, even ubuntu systen, it still come out.

**Comment by wenboweiwb** [comment]
> do you have some solution about it?  I also face the totally some error, and i test several different version of cuda, torch, nvidia-driver, even ubuntu systen, it still come out.

I have tested different combinations of cuda versions, PyTorch versions and Ubuntu versions. For now, I guess this is because your installed cuda version for your system is different from the cuda version used in your Conda environment. It seems like when I run my code in the same cuda version, there is another issue occurs, which is as follows:

return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: cuDNN error: CUDNN_STATUS_EXECUTION_FAILED

It appears once but disappears after I restart my server.

I'm trying to fix this issue by installing PyTorch using Conda (previous I installed PyTorch environment using pip). Different installation methods like pip or Conda may introduce more complex errors I guess. Anyway, any updates, please let me know. Thanks!

**Comment by dunyang97** [comment]
follow your succseeful experience, now i have tried to use the same version of system cuda and env cuda, it seems this error still happen and occured early.

**Comment by dunyang97** [comment]
do you have some updata methd and do you know when this error first come? i am also unclear when this error first happen and which part cause it. It seems i updated my ubuntu syetem and use some new python-package, and it suddenly happen.

**Comment by wenboweiwb** [comment]
> do you have some updata methd and do you know when this error first come? i am also unclear when this error first happen and which part cause it. It seems i updated my ubuntu syetem and use some new python-package, and it suddenly happen.

Yes, it just suddenly happened. I found a common thing that I did every time while the system was running, which is opening the browser. Then, I close the browser and run the code again. It works without any issue! But next time I open the browser, it report the same issue. So, maybe the true reason is the cuda memory assignment. When you open other apps that will use your gpu memory, this may cause this issue. Try it in your own machine and see what is going on. If you made any progress, please let me know thanks.
